### PR TITLE
Generate PlatformManifest and package conflict overrides for Microsoft.AspNetCore.App

### LINKED ISSUE #3609: Handle conflicts between a ref-only Microsoft.AspNetCore.App and individual packages
### Problem

Follow-up to #3608. By making Microsoft.AspNetCore.App a ref-only package, we introduce the possibility of conflicts with other packages, like Microsoft.Extensions.Logging. Both packages would contain assemblies with the same name and version. The .NET Core SDK may not automatically be able to resolve these conflicts, so we need to provide additional information to the conflict resolver on how to handle this scenario.

### Proposed solution

Re-use the mechanism that Microsoft.NETCore.App introduced in 2.0 when moving from granular System.\* packages to a ref-only package.

1. Include a platform manifest (for `dotnet publish` conflict resolution)
1. Supply `PackageConflictOverrides` MSBuild metadata for the conflict resolution task

### Requirements

1. A project using `netcoreapp3.0` and Microsoft.AspNetCore.App should be able to reference the packages and projects which use Microsoft.Extensions.Logging < 3.0.0.
1. A project using `netcoreapp3.0` and Microsoft.AspNetCore.App should build without errors when Microsoft.Extensions.Logging 3.0.0 is referenced (even though its inclusion should be unnecessary)
1. When packages formerly referenced from Microsoft.AspNetCore.App < 3.0 end up in a `netcoreapp3.0` project, they should not end up in publish output for framework dependent apps.
1. When packages formerly referenced from Microsoft.AspNetCore.App < 3.0 end up in a `netcoreapp3.0` project, they should not replace the version in publish output which will come from the runtime-specific version of Microsoft.AspNetCore.App (see https://github.com/aspnet/AspNetCore/issues/3607)

### PR DESCRIPTION
**Pull request by natemcmaster**
This generates additional metadata for the .NET Core SDK to resolve conflicts between Microsoft.AspNetCore.App and PackageReferences which contain the same assemblies.

Resolves https://github.com/aspnet/AspNetCore/issues/3609

Examples of the build output created:

[Microsoft.AspNetCore.App.props](https://github.com/aspnet/AspNetCore/files/2590924/Microsoft.AspNetCore.App.props.txt)
[Microsoft.AspNetCore.App.PlatformManifest.txt](https://github.com/aspnet/AspNetCore/files/2590920/Microsoft.AspNetCore.App.PlatformManifest.txt)

### CODE REVIEW

**Review by dsplaisted — APPROVED** [review]
I didn't look at the implementation, but the build output looks correct.

**Review by muratg — APPROVED** [review]

File: .azure/pipelines/fast-pr-validation.yml [hunk 1]
```diff
@@ -1,7 +1,11 @@
+# Don't run CI for this config
+trigger: none
```
Comment (Eilon) [inline .azure/pipelines/fast-pr-validation.yml]: What's this?

Comment (natemcmaster) [reply .azure/pipelines/fast-pr-validation.yml]: This disables the PR validation trigger when we push to branches. PR validation only needs to run on PRs.

File: src/Framework/test/SharedFxTests.cs [hunk 2]
```diff
@@ -1,29 +1,94 @@
...
             _expectedTfm = "netcoreapp" + TestData.GetPackageVersion().Substring(0, 3);
+            _expectedRid = TestData.GetSharedFxRuntimeIdentifier();
+        }
+
+        [Fact]
+        public void PlatformManifestListsAllFiles()
+        {
+            var platformManifestPath = Path.Combine(TestData.GetManifestOutputDir(), "Microsoft.AspNetCore.App.PlatformManifest.txt");
+            var expectedAssemblies = TestData.GetSharedFxDependencies()
+                .Split(';', StringSplitOptions.RemoveEmptyEntries)
```
Comment (Eilon) [inline src/Framework/test/SharedFxTests.cs]: Is `RemoveEmptyEntries` really needed? Or would it just mask potential bugs where we put blank entries in the metadata?

File: src/Framework/test/SharedFxTests.cs [hunk 3]
```diff
@@ -1,29 +1,94 @@
...
+            AssertEx.FileExists(platformManifestPath);
+
+            var manifestFileLines = File.ReadAllLines(platformManifestPath);
+
+            var actualAssemblies = manifestFileLines
+                .Where(s => !string.IsNullOrEmpty(s))
+                .Select(i =>
+                {
+                    var fileName = i.Split('|')[0];
+                    return fileName.EndsWith(".dll")
```
Comment (Eilon) [inline src/Framework/test/SharedFxTests.cs]: Pass in string comparer. This can fail on other cultures.

**Review by Eilon — APPROVED** [review]
A few small comments/questions.

File: src/Framework/test/SharedFxTests.cs (hunk 2 above)
Comment (natemcmaster) [reply src/Framework/test/SharedFxTests.cs]: Not in this case. I'm only using remove empty entries on TestData.GetSharedFxDependencies(), which is test data generated in the unit test csproj file. The data isn't used in the product.

**Pull request merged**